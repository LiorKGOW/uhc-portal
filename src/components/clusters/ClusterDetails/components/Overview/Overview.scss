.ocm-c-overview-details__card,
.ocm-c-overview-subscription-settings__card {
  &--body {
    .pf-c-description-list {
      margin-top: var(--pf-global--spacer--md);
      --pf-c-description-list__group--RowGap: var(--pf-global--gutter--sm);
      dd button {
        margin-top: var(--pf-global--spacer--lg);
      }
      dl.pf-l-stack dt,
      .autoscale-data-t {
        color: var(--pf-global--Color--400);
        font-size: var(--pf-global--FontSize--sm);
        font-weight: var(--pf-global--FontWeight--bold);
        margin-right: var(--pf-global--spacer--sm);
        text-transform: none;
      }
      .nodes-hint {
        margin-left: var(--pf-global--spacer--sm);
      }
    }
  }
}

// TODO: refactor the global margin-top for pf-c-card
#overviewTabContent {
  padding-top: 24px;

  .pf-c-card {
    margin-top: 0 !important;
  }
}
