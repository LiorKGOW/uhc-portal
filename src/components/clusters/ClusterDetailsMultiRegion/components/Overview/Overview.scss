.ocm-c-overview-details__card,
.ocm-c-overview-subscription-settings__card {
  &--body {
    .pf-v5-c-description-list {
      margin-top: var(--pf-v5-global--spacer--md);
      --pf-v5-c-description-list__group--RowGap: var(--pf-v5-global--gutter--sm);
      dl.pf-v5-l-stack dt,
      .autoscale-data-t {
        color: var(--pf-v5-global--secondary-color--100);
        font-size: var(--pf-v5-global--FontSize--sm);
        font-weight: var(--pf-v5-global--FontWeight--bold);
        margin-right: var(--pf-v5-global--spacer--sm);
        text-transform: none;
      }
      .nodes-hint {
        margin-left: var(--pf-v5-global--spacer--sm);
      }
    }
  }
}

// TODO: refactor the global margin-top for pf-v5-c-card
#overviewTabContent {
  padding-top: 24px;

  .pf-v5-c-card {
    margin-top: 0 !important;
  }
}
