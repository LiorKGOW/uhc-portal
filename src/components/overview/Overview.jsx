import React, { Component } from 'react';
import PropTypes from 'prop-types';

import {
  PageSection,
  CardTitle,
  CardBody,
  EmptyState,
  EmptyStateBody,
  Card,
  Grid,
  GridItem,
} from '@patternfly/react-core';

import { PageHeader, PageHeaderTitle } from '@redhat-cloud-services/frontend-components/PageHeader';
import Spinner from '@redhat-cloud-services/frontend-components/Spinner';

import ConnectedModal from '../common/Modal/ConnectedModal';
import SmallClusterChart from '../clusters/common/ResourceUsage/SmallClusterChart';
import OverviewEmptyState from './OverviewEmptyState';
import ExpiredTrialsCard from './ExpiredTrialsCard';
import ClustersWithIssuesTableCard from './ClustersWithIssuesTableCard';
import CostCard from './CostCard';
import EditSubscriptionSettingsDialog from '../clusters/common/EditSubscriptionSettingsDialog';
import ArchiveClusterDialog from '../clusters/common/ArchiveClusterDialog';
import TopOverviewSection from './TopOverviewSection/TopOverviewSection';
import { createOverviewQueryObject } from '../../common/queryHelpers';
import Unavailable from '../common/Unavailable';
import InsightsAdvisorCard from './InsightsAdvisorCard/InsightsAdvisorCard';

class Overview extends Component {
  componentDidMount() {
    const {
      summaryDashboard,
      getSummaryDashboard,
      unhealthyClusters,
      getUnhealthyClusters,
      getUserAccess,
      viewOptions,
      fetchInsightsGroups,
      insightsGroups,
      fetchOrganizationInsights,
      insightsOverview,
    } = this.props;
    document.title = 'Overview | Red Hat OpenShift Cluster Manager';

    if (!summaryDashboard.fulfilled && !summaryDashboard.pending) {
      getSummaryDashboard();
    }

    if (!unhealthyClusters.fulfilled && !unhealthyClusters.pending) {
      getUnhealthyClusters(createOverviewQueryObject(viewOptions));
    }

    if (!insightsGroups.pending && !insightsGroups.fulfilled) {
      fetchInsightsGroups();
    }

    if (!insightsOverview.pending && !insightsOverview.fulfilled) {
      fetchOrganizationInsights();
    }
    getUserAccess({ type: 'OCP' });
  }

  render() {
    const {
      summaryDashboard,
      unhealthyClusters,
      viewOptions,
      invalidateSubscriptions,
      totalClusters,
      totalConnectedClusters,
      totalUnhealthyClusters,
      totalCPU,
      usedCPU,
      totalMem,
      usedMem,
      upToDate,
      upgradeAvailable,
      insightsGroups,
      insightsOverview,
      userAccess,
    } = this.props;

    const isError = (summaryDashboard.error || unhealthyClusters.error);
    const isPending = (!summaryDashboard.fulfilled || summaryDashboard.pending
       || !unhealthyClusters.fulfilled || unhealthyClusters.pending);
    // TODO: should show only when at least one cluster is connected and sends Insights
    const showInsightsAdvisorWidget = insightsOverview.fulfilled && insightsOverview.overview;

    if (isError) {
      let errorSource;
      if (summaryDashboard.error) {
        errorSource = summaryDashboard;
      } else {
        errorSource = unhealthyClusters;
      }
      const { errorMessage, errorCode, operationID } = errorSource;
      const response = { errorMessage, errorCode, operationID };
      return (
        <Unavailable
          response={response}
        />
      );
    }

    // Show spinner if while waiting for responses.
    if (isPending && !isError) {
      return (
        <EmptyState>
          <EmptyStateBody>
            <Spinner centered />
          </EmptyStateBody>
        </EmptyState>
      );
    }

    // Revert to an "empty" state if there are no clusters to show.
    if (summaryDashboard.fulfilled && !totalClusters) {
      return (
        <PageSection>
          <OverviewEmptyState />
        </PageSection>
      );
    }

    return (
      <>
        <PageHeader>
          <PageHeaderTitle title="Overview" className="page-title" />
        </PageHeader>
        <PageSection>
          <Grid hasGutter id="overview-grid">
            <TopOverviewSection
              isError={summaryDashboard.error}
              totalClusters={totalClusters}
              totalUnhealthyClusters={totalUnhealthyClusters}
              totalConnectedClusters={totalConnectedClusters}
              totalCPU={totalCPU}
              usedCPU={usedCPU}
              totalMem={totalMem}
              usedMem={usedMem}
            />
            {totalConnectedClusters > 0 && (
            <GridItem md={6} sm={12}>
              <ClustersWithIssuesTableCard
                unhealthyClusters={unhealthyClusters}
                viewOptions={viewOptions}
              />
            </GridItem>
            )}
            {showInsightsAdvisorWidget && (
              <GridItem md={6} sm={12}>
                <InsightsAdvisorCard
                  overview={insightsOverview.overview}
                  groups={insightsGroups.groups}
                />
              </GridItem>
            )}
            <GridItem md={6} sm={12}>
              <Card className="clusters-overview-card">
                <CardTitle>
                  Telemetry
                </CardTitle>
                <CardBody>
                  {!totalConnectedClusters && !totalClusters ? (
                    <EmptyState>
                      <EmptyStateBody>
                        No data available
                      </EmptyStateBody>
                    </EmptyState>
                  ) : (
                    <SmallClusterChart
                      donutId="connected_clusters_donut"
                      used={totalConnectedClusters}
                      total={totalClusters}
                      availableTitle="Not checking in"
                      usedTitle="Connected"
                      unit="clusters"
                    />
                  )}
                </CardBody>
              </Card>
            </GridItem>
            {userAccess.fulfilled && userAccess.data !== undefined && userAccess.data === true && (
              <GridItem md={6} sm={12}>
                <CostCard />
              </GridItem>
            )}
            <GridItem md={6} sm={12}>
              <Card className="clusters-overview-card">
                <CardTitle>
                  Update status
                </CardTitle>
                <CardBody>
                  {!upgradeAvailable.value && !upToDate.value
                    ? (
                      <EmptyState>
                        <EmptyStateBody>
                          No data available
                        </EmptyStateBody>
                      </EmptyState>
                    )
                    : (
                      <SmallClusterChart
                        donutId="update_available_donut"
                        used={upToDate.value}
                        total={upgradeAvailable.value + upToDate.value}
                        unit="clusters"
                        availableTitle="Update available"
                        usedTitle="Up-to-date"
                      />
                    )}
                </CardBody>
              </Card>
            </GridItem>
            <GridItem span={12}>
              <ExpiredTrialsCard />
            </GridItem>
          </Grid>
          <ConnectedModal
            ModalComponent={EditSubscriptionSettingsDialog}
            onClose={invalidateSubscriptions}
          />
          <ConnectedModal ModalComponent={ArchiveClusterDialog} onClose={invalidateSubscriptions} />
        </PageSection>
      </>
    );
  }
}

Overview.propTypes = {
  getSummaryDashboard: PropTypes.func.isRequired,
  getUserAccess: PropTypes.func.isRequired,
  invalidateSubscriptions: PropTypes.func.isRequired,
  summaryDashboard: PropTypes.object.isRequired,
  getUnhealthyClusters: PropTypes.func.isRequired,
  unhealthyClusters: PropTypes.shape({
    clusters: PropTypes.array,
    error: PropTypes.bool,
    pending: PropTypes.bool,
    fulfilled: PropTypes.bool,
  }).isRequired,
  viewOptions: PropTypes.shape({
    currentPage: PropTypes.number,
    pageSize: PropTypes.number,
    totalCount: PropTypes.number,
    totalPages: PropTypes.number,
  }).isRequired,
  totalClusters: PropTypes.number.isRequired,
  totalConnectedClusters: PropTypes.number.isRequired,
  totalUnhealthyClusters: PropTypes.number.isRequired,
  totalCPU: PropTypes.object.isRequired,
  usedCPU: PropTypes.object.isRequired,
  totalMem: PropTypes.object.isRequired,
  usedMem: PropTypes.object.isRequired,
  upToDate: PropTypes.object.isRequired,
  upgradeAvailable: PropTypes.object.isRequired,
  fetchInsightsGroups: PropTypes.func.isRequired,
  fetchOrganizationInsights: PropTypes.func.isRequired,
  insightsGroups: PropTypes.object.isRequired,
  insightsOverview: PropTypes.object.isRequired,
  userAccess: PropTypes.shape({
    data: PropTypes.bool,
    pending: PropTypes.bool,
    fulfilled: PropTypes.bool,
  }).isRequired,
};

export default Overview;
