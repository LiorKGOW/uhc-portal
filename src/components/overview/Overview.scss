@import '~@patternfly/patternfly/sass-utilities/_all';

$ocm-c-marketing-banner--breakpoint-map: build-breakpoint-map();

.ocm-c-marketing-banner {
  --ocm-m-marketing-page--RowGap: var(--pf-global--spacer--xl);
  --ocm-m-marketing-page--ColumnGap: var(--pf-global--spacer--xl);
  --ocm-m-marketing-page--section--PaddingTop: var(--pf-global--spacer--xl);
  --ocm-m-marketing-page--section--PaddingRight: var(--pf-global--spacer--2xl);
  --ocm-m-marketing-page--section--PaddingBottom: var(--pf-global--spacer--xl);
  --ocm-m-marketing-page--section--PaddingLeft: var(--pf-global--spacer--2xl);

  .ocm-l-autofit {
    --ocm-l-autofit--m-gutter--RowGap: var(--ins-m-marketing-page--RowGap);
    --ocm-l-autofit--m-gutter--ColumnGap: var(--ins-m-marketing-page--ColumnGap);
  }

  .pf-c-page__main-section {
    --pf-c-page__main-section--PaddingTop: var(--ins-m-marketing-page--section--PaddingTop);
    --pf-c-page__main-section--PaddingRight: var(--ins-m-marketing-page--section--PaddingRight);
    --pf-c-page__main-section--PaddingBottom: var(--ins-m-marketing-page--section--PaddingBottom);
    --pf-c-page__main-section--PaddingLeft: var(--ins-m-marketing-page--section--PaddingLeft);
  }

  .ocm-c-marketing-banner {
    --ocm-c-marketing-banner--PaddingTop: var(--pf-global--spacer--2xl);
    --ocm-c-marketing-banner--PaddingBottom: var(--pf-global--spacer--2xl);
  }

  a {
    color: #000;
    text-decoration: none;
  }

  .pf-l-grid {
    --pf-l-grid--m-gutter--GridGap: var(--pf-global--spacer--xl);
  }
}



.dashboard-emptystate-card .pf-c-card__body {
  a {
    display: block;
    margin-top: 16px;
  }

  .pf-c-button {
    padding-left: 0;
  }

  .pf-c-button > .pf-c-button__icon {
    float: right;
    margin-left: 5px;
  }
}

.dashboard-emptystate-card{
  max-width: 570px;
}

#dashboard-emptystate-osd {
  padding-bottom: var(--pf-global--spacer--lg);
}

#overview-emptystate-get-productive .flex-pair{
  div{
    margin-bottom: var(--pf-global--spacer--lg);
    @media (max-width: 480px) {
      width: 338px;
    }
    @media (min-width: 481px) and (max-width: 768px) {
      width: 520px;
    }
    @media (min-width: 779px) and (max-width: 1199px){
      width: 463px;
    }
  }
  div:first-child{
      @media (min-width: 1023px) {
      margin-right: var(--pf-global--spacer--lg);
    }
  }
}

#emptystate-learn-more {
  display: flex;
  flex-wrap: wrap;

  iframe {
    min-height: 375px;
  }

  .pf-c-list{
   list-style: none;
  }

  div:first-child{
    @media (max-width: 769px) {
      margin-bottom: var(--pf-global--spacer--lg);
    }
    @media (min-width: 737px) {
      margin-right: var(--pf-global--spacer--lg);
    }
  }
}

#request-information-btn {
  text-decoration: none;
  button {
    display: block;
    margin-top: var(--pf-global--spacer--md);
  }
}

#create-cluster-card-image {
  height: 12em;
  width: 18em;
  padding-right: 1em;
  padding-top: 1em;
}