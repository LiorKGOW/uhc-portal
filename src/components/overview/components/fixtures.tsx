import React from 'react';

import SomeLogo from '~/styles/images/RedHatOpenShiftGitOpsLogo.svg';

import { DRAWER_PANEL_CONTENT, DrawerPanelContentNode } from './common/DrawerPanelContent';
import PRODUCT_CARD_LOGOS from './common/ProductCardLogos';
import { FEATURED_PRODUCTS_CARDS } from './FeaturedProductsCards/FeaturedProductsCards';

const BASIC_PRODUCT_CARD_PROPS = {
  title: 'example title',
  description: 'example description',
  logo: 'product-overview-card__logo',
  labelText: 'example labelText',
  drawerPanelContent: {
    head: <div data-testid="test-head-div">example drawer panel content head</div>,
    body: <div data-testid="test-body-div">example drawer panel content body</div>,
  },
  openLearnMore: jest.fn(),
  isSelected: false,
  dataTestId: 'some-id',
};

const PRODUCT_CARD_TEST_CASES = {
  UNSELECTED: {
    ...BASIC_PRODUCT_CARD_PROPS,
  },

  SELECTED: {
    ...BASIC_PRODUCT_CARD_PROPS,
    isSelected: true,
  },

  WITHOUT_LABEL_TEXT: {
    ...BASIC_PRODUCT_CARD_PROPS,
    labelText: undefined,
  },
};

const BASIC_DRAWER_PANEL_PROPS = {
  children: <div data-testid="children-of-drawer-panel">some children</div>,
  content: {
    head: <div data-testid="test-head-div">example drawer panel content head</div>,
    body: <div data-testid="test-body-div">example drawer panel content body</div>,
  },
  isOpen: true,
  onClose: jest.fn(),
};

const DRAWER_PANEL_TEST_CASES = {
  OPENED: { ...BASIC_DRAWER_PANEL_PROPS },
};

const BASIC_RECCOMENDED_OPERATORS_CARDS_PROPS = {
  openLearnMore: jest.fn(),
  selectedCardTitle: '',
};

const RECOMMENDED_OPERATORS_CARDS_TEST_CASES = {
  NON_SELECTED: {
    ...BASIC_RECCOMENDED_OPERATORS_CARDS_PROPS,
  },
  GITOPS_SELECTED: {
    ...BASIC_RECCOMENDED_OPERATORS_CARDS_PROPS,
    selectedCardTitle: PRODUCT_CARD_LOGOS.gitops.title,
  },
};

type RecommendedOperatorsCardsNode = {
  title: string;
  description: string;
  logo: string | undefined;
  labelText?: string;
  drawerPanelContent?: DrawerPanelContentNode;
};

const RECOMMENDED_OPERATORS_CARDS_DATA: RecommendedOperatorsCardsNode[] = [
  {
    ...PRODUCT_CARD_LOGOS.gitops,
    description:
      'Integrate git repositories, continuous integration/continuous delivery (CI/CD) tools, and Kubernetes.',
    labelText: 'Free',
    drawerPanelContent: DRAWER_PANEL_CONTENT.gitops,
  },
  {
    ...PRODUCT_CARD_LOGOS.pipelines,
    description:
      'Automate your application delivery using a continuous integration and continuous deployment (CI/CD) framework.',
    labelText: 'Free',
    drawerPanelContent: DRAWER_PANEL_CONTENT.pipelines,
  },
  {
    ...PRODUCT_CARD_LOGOS.serviceMesh,
    description: 'Connect, manage, and observe microservices-based applications in a uniform way.',
    labelText: 'Free',
    drawerPanelContent: DRAWER_PANEL_CONTENT.serviceMesh,
  },
];

const BASIC_FEATURED_PRODUCTS_CARDS_PROPS = {
  openLearnMore: jest.fn(),
  selectedCardTitle: '',
};

const FEATURED_PRODUCTS_CARDS_TEST_CASES = {
  NON_SELECTED: {
    ...BASIC_FEATURED_PRODUCTS_CARDS_PROPS,
  },
  ADVANCED_CLUSTER_SECURITY_SELECTED: {
    ...BASIC_FEATURED_PRODUCTS_CARDS_PROPS,
    selectedCardTitle: PRODUCT_CARD_LOGOS.advancedClusterSecurity.title,
  },
};

const GENERAL_DRAWER_PANEL_HEAD_BASIC = {
  title: 'some title',
  logo: SomeLogo,
};

const GENERAL_DRAWER_PANEL_HEAD_WITH_LINK = {
  ...GENERAL_DRAWER_PANEL_HEAD_BASIC,
  trialButtonLink: 'https://www.redhat.com/en',
};

export {
  PRODUCT_CARD_TEST_CASES,
  DRAWER_PANEL_TEST_CASES,
  RECOMMENDED_OPERATORS_CARDS_TEST_CASES,
  RECOMMENDED_OPERATORS_CARDS_DATA,
  FEATURED_PRODUCTS_CARDS_TEST_CASES,
  FEATURED_PRODUCTS_CARDS,
  GENERAL_DRAWER_PANEL_HEAD_BASIC,
  GENERAL_DRAWER_PANEL_HEAD_WITH_LINK,
};
