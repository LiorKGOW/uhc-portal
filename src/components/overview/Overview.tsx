import './Overview.scss';
import React from 'react';
import { Button, Title, Label, Flex, FlexItem, PageSection } from '@patternfly/react-core';
import { Link } from 'react-router-dom';
import ExternalLink from '~/components/common/ExternalLink';
import useAnalytics from '~/hooks/useAnalytics';
import { trackEvents } from '~/common/analytics';
import { ProductBanner, ProductBannerProps } from '../common/ProductBanner';
import docLinks from '../../common/installLinks.mjs';
import {
  ListTextLabelLinkCard,
  ListTextLabelLinkCardProps,
} from '../common/ListTextLabelLinkCard/ListTextLabelLinkCard';
import OpenShiftProductIcon from '../../styles/images/OpenShiftProductIcon.svg';
import { OfferingCard } from './OfferingCard/OfferingCard';
import { AppPage } from '../App/AppPage';

const linkTextLabelLinkCardContents: ListTextLabelLinkCardProps = {
  cardClassName: 'pf-v5-u-mb-lg',
  textLabelLinkItems: [
    {
      listItemText: 'Using Red Hat OpenShift Cluster Manager to work with your OpenShift clusters',
      listItemLabel: <Label color="gold">Documentation</Label>,
      listItemLink: (
        <ExternalLink href={docLinks.OCM_DOCS_MANAGING_CLUSTERS}>Learn More</ExternalLink>
      ),
    },
    {
      listItemText: 'OpenShift Serverless overview',
      listItemLabel: <Label color="gold">Documentation</Label>,
      listItemLink: <ExternalLink href={docLinks.SERVERLESS_ABOUT}>Learn More</ExternalLink>,
    },
    {
      listItemText: 'Understanding Service Mesh',
      listItemLabel: <Label color="gold">Documentation</Label>,
      listItemLink: <ExternalLink href={docLinks.SERVICE_MESH_ABOUT}>Learn More</ExternalLink>,
    },
    {
      listItemText: 'About OpenShift Virtualization',
      listItemLabel: <Label color="gold">Documentation</Label>,
      listItemLink: <ExternalLink href={docLinks.VIRT_ABOUT}>Learn More</ExternalLink>,
    },
  ],
};

const openshiftBannerContents: ProductBannerProps = {
  icon: <img src={OpenShiftProductIcon} alt="OpenShift product icon" />,
  learnMoreLink: (
    <ExternalLink href={docLinks.WHAT_IS_OPENSHIFT}>Learn more about OpenShift</ExternalLink>
  ),
  title: 'Get started with OpenShift',
  text: (
    <>
      Focus on work that matters with the industry&#39;s leading hybrid cloud application platform
      powered by Kubernetes.
      <br />
      Develop, modernize, deploy, run, and manage your applications faster and easier.
    </>
  ),
};

function OverviewEmptyState() {
  const track = useAnalytics();
  const createClusterURL = '/create';
  return (
    <AppPage>
      <ProductBanner
        icon={openshiftBannerContents.icon}
        learnMoreLink={openshiftBannerContents.learnMoreLink}
        title={openshiftBannerContents.title}
        text={openshiftBannerContents.text}
      />
      <PageSection>
        <Title size="xl" headingLevel="h2" className="pf-v5-u-mt-lg">
          Featured OpenShift cluster types
        </Title>
        <Flex className="pf-v5-u-mb-lg">
          <FlexItem className="pf-v5-u-pt-md">
            <OfferingCard offeringType="RHOSD" />
          </FlexItem>
          <FlexItem className="pf-v5-u-pt-md">
            <OfferingCard offeringType="AWS" />
          </FlexItem>
          <FlexItem className="pf-v5-u-pt-md">
            <OfferingCard offeringType="Azure" />
          </FlexItem>
          <FlexItem className="pf-v5-u-pt-md">
            <OfferingCard offeringType="RHOCP" />
          </FlexItem>
          <FlexItem className="pf-v5-u-pt-md">
            <OfferingCard offeringType="RHOIBM" />
          </FlexItem>
          <FlexItem className="pf-v5-u-pt-md">
            <OfferingCard offeringType="DEVSNBX" />
          </FlexItem>
        </Flex>
        <Button
          variant="link"
          component={(props) => (
            <Link
              {...props}
              onClick={() => {
                track(trackEvents.CreateCluster, {
                  url: createClusterURL,
                  path: window.location.pathname,
                });
              }}
              data-testid="create-cluster"
              to={createClusterURL}
            />
          )}
        >
          View all OpenShift cluster types
        </Button>
        <Title size="xl" headingLevel="h2" className="pf-v5-u-mt-lg pf-v5-u-mb-lg">
          Recommended Content
        </Title>
        <ListTextLabelLinkCard {...linkTextLabelLinkCardContents} />
        <ExternalLink href="/openshift/learning-resources">
          Browse all OpenShift learning resources
        </ExternalLink>
      </PageSection>
    </AppPage>
  );
}

export default OverviewEmptyState;
