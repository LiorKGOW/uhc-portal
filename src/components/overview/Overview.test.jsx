import React from 'react';
import { shallow } from 'enzyme';
import Overview from './Overview';
import { clustersWithIssues } from './Overview.fixtures';

const dashboardState = {
  summary: {
    fulfilled: true,
    pending: false,
    metrics: {
      clusters_total: [
        { value: 30 },
      ],
      connected_clusters_total: [
        { value: 10 },
      ],
      unhealthy_clusters_total: [
        { value: 5 },
      ],
      sum_total_cpu: [
        { value: 150 },
      ],
      sum_used_cpu: [
        { value: 10 },
      ],
      sum_total_memory: [
        { value: 100 },
      ],
      sum_used_memory: [
        { value: 10 },
      ],
      clusters_up_to_date_total: [
        { value: 10 },
      ],
      clusters_upgrade_available_total: [
        { value: 1 },
      ],
    },
    error: false,
  },
  unhealthyClusters: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: true,
    valid: false,
    clusters: clustersWithIssues,
  },
  insightsOverview: {},
  insightsGroups: {},
};

describe('<Overview />', () => {
  let getSummaryDashboard;
  let getUnhealthyClusters;
  let wrapper;
  let viewOptions;
  let fetchInsightsGroups;
  let fetchOrganizationInsights;
  beforeAll(() => {
    getSummaryDashboard = jest.fn();
    getUnhealthyClusters = jest.fn();
    fetchInsightsGroups = jest.fn();
    fetchOrganizationInsights = jest.fn();
    viewOptions = {
      currentPage: 1,
      pageSize: 1,
      totalCount: 1,
      totalPages: 1,
    };
    wrapper = shallow(
      <Overview
        getSummaryDashboard={getSummaryDashboard}
        invalidateSubscriptions={jest.fn()}
        getUnhealthyClusters={getUnhealthyClusters}
        unhealthyClusters={dashboardState.unhealthyClusters}
        summaryDashboard={dashboardState.summary}
        totalClusters={dashboardState.summary.metrics.clusters_total[0].value}
        totalConnectedClusters={dashboardState.summary.metrics.connected_clusters_total[0].value}
        totalUnhealthyClusters={dashboardState.summary.metrics.unhealthy_clusters_total[0].value}
        totalCPU={dashboardState.summary.metrics.sum_total_cpu[0]}
        usedCPU={dashboardState.summary.metrics.sum_used_cpu[0]}
        totalMem={dashboardState.summary.metrics.sum_total_memory[0]}
        usedMem={dashboardState.summary.metrics.sum_used_memory[0]}
        upToDate={dashboardState.summary.metrics.clusters_up_to_date_total[0]}
        upgradeAvailable={dashboardState.summary.metrics.clusters_upgrade_available_total[0]}
        viewOptions={viewOptions}
        fetchInsightsGroups={fetchInsightsGroups}
        fetchOrganizationInsights={fetchOrganizationInsights}
        insightsOverview={dashboardState.insightsOverview}
        insightsGroups={dashboardState.insightsGroups}
      />,
    );
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
