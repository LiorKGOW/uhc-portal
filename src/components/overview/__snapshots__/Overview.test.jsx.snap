// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overview /> renders correctly 1`] = `
<Fragment>
  <PageHeader>
    <PageHeaderTitle
      className="page-title"
      title="Overview"
    />
  </PageHeader>
  <PageSection>
    <Grid
      hasGutter={true}
      id="overview-grid"
    >
      <TopOverviewSection
        isError={false}
        totalCPU={
          Object {
            "value": 150,
          }
        }
        totalClusters={30}
        totalConnectedClusters={10}
        totalMem={
          Object {
            "value": 100,
          }
        }
        totalUnhealthyClusters={5}
        usedCPU={
          Object {
            "value": 10,
          }
        }
        usedMem={
          Object {
            "value": 10,
          }
        }
      />
      <GridItem
        md={6}
        sm={12}
      >
        <Connect(ClustersWithIssuesTableCard)
          unhealthyClusters={
            Object {
              "clusters": Array [
                Object {
                  "api": Object {
                    "url": "https://api.test-liza.sdev.devshift.net:6443",
                  },
                  "canDelete": true,
                  "canEdit": true,
                  "cloud_provider": Object {
                    "href": "/api/clusters_mgmt/v1/cloud_providers/aws",
                    "id": "aws",
                    "kind": "CloudProviderLink",
                  },
                  "cluster_admin_enabled": false,
                  "console": Object {
                    "url": "https://console-openshift-console.apps.test-liza.sdev.devshift.net",
                  },
                  "creation_timestamp": "2019-03-26T15:18:39.06783Z",
                  "display_name": "test-liza",
                  "dns": Object {
                    "base_domain": "sdev.devshift.net",
                  },
                  "external_id": "9f50940b-fba8-4c59-9c6c-d64284b2026d",
                  "flavour": Object {
                    "href": "/api/clusters_mgmt/v1/flavours/osd-4",
                    "id": "osd-4",
                    "kind": "FlavourLink",
                  },
                  "health_state": "unhealthy",
                  "href": "/api/clusters_mgmt/v1/clusters/1IztzhAGrbjtKkMbiPewJanhTXk",
                  "id": "1IztzhAGrbjtKkMbiPewJanhTXk",
                  "kind": "Cluster",
                  "load_balancer_quota": 0,
                  "managed": true,
                  "metrics": Object {
                    "cpu": Object {
                      "total": Object {
                        "unit": "",
                        "value": 16,
                      },
                      "updated_timestamp": "2019-04-28T14:23:18Z",
                      "used": Object {
                        "unit": "",
                        "value": 3.995410922987096,
                      },
                    },
                    "critical_alerts_firing": 5,
                    "memory": Object {
                      "total": Object {
                        "unit": "B",
                        "value": 82293346304,
                      },
                      "updated_timestamp": "2019-04-28T14:23:19Z",
                      "used": Object {
                        "unit": "B",
                        "value": 16546058240,
                      },
                    },
                    "nodes": Object {
                      "compute": 4,
                      "master": 3,
                      "total": 7,
                    },
                    "sockets": Object {
                      "total": Object {
                        "unit": "",
                        "value": 0,
                      },
                      "updated_timestamp": "0001-01-01T00:00:00Z",
                      "used": Object {
                        "unit": "",
                        "value": 0,
                      },
                    },
                    "storage": Object {
                      "total": Object {
                        "unit": "B",
                        "value": 0,
                      },
                      "updated_timestamp": "0001-01-01T00:00:00Z",
                      "used": Object {
                        "unit": "B",
                        "value": 0,
                      },
                    },
                  },
                  "multi_az": false,
                  "name": "test-liza",
                  "network": Object {
                    "machine_cidr": "10.0.0.0/16",
                    "pod_cidr": "10.128.0.0/14",
                    "router_shards": Object {
                      "href": "/api/clusters_mgmt/v1/clusters/1IztzhAGrbjtKkMbiPewJanhTXk/router_shards",
                      "id": "1IztzhAGrbjtKkMbiPewJanhTXk",
                      "kind": "RouterShardListLink",
                    },
                    "service_cidr": "172.30.0.0/16",
                  },
                  "nodes": Object {
                    "compute": 2,
                    "master": 3,
                    "total": 5,
                  },
                  "region": Object {
                    "href": "/api/clusters_mgmt/v1/cloud_providers/aws/regions/us-east-1",
                    "id": "us-east-1",
                    "kind": "CloudRegionLink",
                  },
                  "state": "ready",
                  "storage_quota": Object {
                    "unit": "B",
                    "value": 107374182400,
                  },
                  "subscription": Object {
                    "id": "foo",
                  },
                  "version": Object {
                    "href": "/api/clusters_mgmt/v1/versions/openshift-v4.0-latest",
                    "id": "openshift-v4.0-latest",
                    "kind": "VersionLink",
                  },
                },
              ],
              "error": false,
              "errorMessage": "",
              "fulfilled": true,
              "pending": false,
              "valid": false,
            }
          }
          viewOptions={
            Object {
              "currentPage": 1,
              "pageSize": 1,
              "totalCount": 1,
              "totalPages": 1,
            }
          }
        />
      </GridItem>
      <GridItem
        md={6}
        sm={12}
      >
        <Card
          className="clusters-overview-card"
        >
          <CardTitle>
            Telemetry
          </CardTitle>
          <CardBody>
            <SmallClusterChart
              availableTitle="Not checking in"
              donutId="connected_clusters_donut"
              total={30}
              unit="clusters"
              used={10}
              usedTitle="Connected"
            />
          </CardBody>
        </Card>
      </GridItem>
      <GridItem
        md={6}
        sm={12}
      >
        <Card
          className="clusters-overview-card"
        >
          <CardTitle>
            Update status
          </CardTitle>
          <CardBody>
            <SmallClusterChart
              availableTitle="Update available"
              donutId="update_available_donut"
              total={11}
              unit="clusters"
              used={10}
              usedTitle="Up-to-date"
            />
          </CardBody>
        </Card>
      </GridItem>
      <GridItem
        span={12}
      >
        <Connect(ExpiredTrialsCard) />
      </GridItem>
    </Grid>
    <Connect(ConnectedModal)
      ModalComponent={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(EditSubscriptionSettingsDialog)",
          "modalName": "edit-subscription-settings",
          "type": [Function],
        }
      }
      onClose={[MockFunction]}
    />
    <Connect(ConnectedModal)
      ModalComponent={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(ArchiveClusterDialog)",
          "modalName": "archive-cluster",
          "type": [Function],
        }
      }
      onClose={[MockFunction]}
    />
  </PageSection>
</Fragment>
`;
